src/write.rs

```diff
@@ -433,14 +408,10 @@ impl<W: Write + Seek> GnuBuilder<W> {
         } else {
             self.short_names.contains(header.identifier())
         };
-        if !has_name {
-            let msg = format!(
-                "Identifier {:?} was not in the list of \
-                 identifiers passed to GnuBuilder::new()",
-                String::from_utf8_lossy(header.identifier())
-            );
-            return Err(Error::new(ErrorKind::InvalidInput, msg));
-        }
+
+        ensure!(has_name,
+            "Identifier `{}` was not in the list of identifiers passed to GnuBuilder::new()",
```

Comment:
```suggestion
            "Identifier {:?} was not in the list of identifiers passed to GnuBuilder::new()",
```

This escapes characters like `\0` and `\n`. Otherwise they would be invisible.

Reply:
Well tbh if thats the intent we should just print the bytes and not try to use a string.

The end user _gave us bytes_ in the first place